### ISSUE TITLE
Obsolete warning from kernel 

### ISSUE THREAD

**Issue by Chocobo1** [ppp-project/ppp#411]
Hi,
I noticed these messages in system log:
```
kernel: NET: Registered PF_PPPOX protocol family
kernel: pppd uses obsolete (PF_INET,SOCK_PACKET)
```
I'm running linux 6.2.9 and latest ppp from git. And if I'm not mistaken, it originates from here: https://github.com/ppp-project/ppp/blob/760ce18f82670eb81cc186fb792919339a2e2fbe/pppd/plugins/pppoe/if.c#L115-L116

Note that just above there is a compile-time macro: https://github.com/ppp-project/ppp/blob/760ce18f82670eb81cc186fb792919339a2e2fbe/pppd/plugins/pppoe/if.c#L111

But it doesn't seem that macro is defined nor detected anywhere.

**Comment by Chocobo1** [comment]
>@Chocobo1: Have you looked comments here: https://github.com/ppp-project/ppp/pull/372?

Yes, see: https://github.com/ppp-project/ppp/pull/372#issuecomment-1257466619:
>just take the patch and reapply under a different author (I'm fine with it).
@Neustradamus I don't mind you take the changes and submit it.

**Comment by wongsyrone** [comment]
This is from the original rp-pppoe repository: https://salsa.debian.org/dskoll/rp-pppoe/-/blob/master/src/configure.ac#L32

And I fix it via https://github.com/wongsyrone/lede-1/commit/ae833ac9e14a4a874144807b15f01c5d922a54c0

**Comment by Chocobo1** [comment]
>And I fix it via https://github.com/wongsyrone/lede-1/commit/ae833ac9e14a4a874144807b15f01c5d922a54c0

Will you consider to submit a PR to this repo?

**Comment by Chocobo1** [comment]
For anyone interested, here is my fix: https://github.com/Chocobo1/ppp/commit/bbf65165fa6c642ced13e6970b3b3e04cfcff9f5

**Comment by spanevin** [comment]
When the fix is going to be released? It doesn't look like the latest 2.5.0 has it included...

**Comment by Neustradamus** [comment]
@paulusmack: @spanevin has requested a new build, it is possible to do it?
Maybe good to look current PR too before?
